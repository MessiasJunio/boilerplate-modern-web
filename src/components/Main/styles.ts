'use client'

import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #fcedeeff;
  color: #0e394fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`

export const Logo = styled.img`
  width: 25rem;
  margin-bottom: 2rem;
`

export const Title = styled.h1`
  font-size: 2.5rem;
  margin-bottom: 2rem;
`

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
  margin-bottom: 3rem;
`

export const Illustration = styled.img`
  width: min(30rem, 100%);
`

export const IconBox = styled.div`
  display: flex;
  gap: 1rem;
  margin-top: 3rem;
`

export const Icon = styled.img`
  height: 5rem;
  width: 5rem;
`
