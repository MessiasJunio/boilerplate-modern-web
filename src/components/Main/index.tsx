import Link from 'next/link'
import * as S from './styles'

const Main = ({
  title = 'Modern Web',
  description = 'TypeScript, ReactJS, NextJS, Storybook, Styled Components and Jest'
}) => (
  <S.Wrapper>
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
    <S.Illustration
      src="/img/hero-illustration.svg"
      alt="A Robot behind a laptop"
    />
    <S.IconBox>
      <Link href={'https://www.typescriptlang.org/'}>
        <S.Icon
          src="/icons/typescript.svg"
          alt="A typescript programming language"
        />
      </Link>
      <Link href={'https://react.dev/'}>
        <S.Icon src="/icons/react.svg" alt="React framework" />
      </Link>
      <Link href={'https://nextjs.org/'}>
        <S.Icon src="/icons/nextjs.svg" alt="Next.js framework" />
      </Link>
      <Link href={'https://storybook.js.org/'}>
        <S.Icon
          src="/icons/storybook.svg"
          alt="Storybook a frontend workshop"
        />
      </Link>
      <Link href={'https://styled-components.com/'}>
        <S.Icon
          src="/icons/styled-components.svg"
          alt="Styled components CSS-in-JS"
        />
      </Link>
      <Link href={'https://jestjs.io/'}>
        <S.Icon src="/icons/jest.svg" alt="Jest a testing framework" />
      </Link>
    </S.IconBox>
  </S.Wrapper>
)

export default Main
