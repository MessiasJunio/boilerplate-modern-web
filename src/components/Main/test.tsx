import { render, screen } from '@testing-library/react'
import Main from '.'

describe('<Main />', () => {
  it('should render heading', () => {
    const { container } = render(<Main />)

    expect(screen.getByRole('heading', { name: /Modern Web/i }))

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the background color correctly', () => {
    const { container } = render(<Main />)
    expect(container.firstChild).toHaveStyle({
      'background-color': '#fcedeeff'
    })
  })
})
