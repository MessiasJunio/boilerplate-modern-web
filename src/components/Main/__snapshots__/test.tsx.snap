// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render heading 1`] = `
.c0 {
  background-color: #fcedeeff;
  color: #0e394fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.c1 {
  font-size: 2.5rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2rem;
  font-weight: 400;
  margin-bottom: 3rem;
}

.c3 {
  width: min(30rem, 100%);
}

.c4 {
  display: flex;
  gap: 1rem;
  margin-top: 3rem;
}

.c5 {
  height: 5rem;
  width: 5rem;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    Modern Web
  </h1>
  <h2
    class="c2"
  >
    TypeScript, ReactJS, NextJS, Storybook, Styled Components and Jest
  </h2>
  <img
    alt="A Robot behind a laptop"
    class="c3"
    src="/img/hero-illustration.svg"
  />
  <div
    class="c4"
  >
    <a
      href="https://www.typescriptlang.org/"
    >
      <img
        alt="A typescript programming language"
        class="c5"
        src="/icons/typescript.svg"
      />
    </a>
    <a
      href="https://react.dev/"
    >
      <img
        alt="React framework"
        class="c5"
        src="/icons/react.svg"
      />
    </a>
    <a
      href="https://nextjs.org/"
    >
      <img
        alt="Next.js framework"
        class="c5"
        src="/icons/nextjs.svg"
      />
    </a>
    <a
      href="https://storybook.js.org/"
    >
      <img
        alt="Storybook a frontend workshop"
        class="c5"
        src="/icons/storybook.svg"
      />
    </a>
    <a
      href="https://styled-components.com/"
    >
      <img
        alt="Styled components CSS-in-JS"
        class="c5"
        src="/icons/styled-components.svg"
      />
    </a>
    <a
      href="https://jestjs.io/"
    >
      <img
        alt="Jest a testing framework"
        class="c5"
        src="/icons/jest.svg"
      />
    </a>
  </div>
</main>
`;
